src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchClient.kt

```diff
@@ -1,21 +1,58 @@
 package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
 
+import mu.KotlinLogging
+import org.springframework.http.HttpStatus.FORBIDDEN
 import org.springframework.http.MediaType
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
 import org.springframework.web.reactive.function.client.bodyToMono
 import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PagePrisoner
 import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches
 
+private val log = KotlinLogging.logger {}
+
 @Component
 class PrisonerSearchClient(private val prisonerSearchWebClient: WebClient) {
   fun matchPrisoners(prisonerSearchRequest: PrisonerSearchRequest): Mono<PrisonerMatches> {
     return prisonerSearchWebClient
       .post()
       .uri("/match-prisoners")
       .accept(MediaType.APPLICATION_JSON)
-      .bodyValue(prisonerSearchRequest.toRequestBody())
+      .bodyValue(prisonerSearchRequest.toMatchPrisonersRequestBody())
+      .retrieve()
+      .bodyToMono<PrisonerMatches>()
+      .onErrorResume { webClientErrorHandler(it) }
+    // In theory this is a possible way to submit the request on a different thread ... but it can't get the auth token because
+    // `userContext` is on the main request thread, so [WebClientConfiguration#addAuthHeaderFilterFunction] results in a NPE
+    // Needs some thought if we want to do this .... (other concurrency approaches may suffer the same fate if they use a different thread)
+    // .subscribeOn(Schedulers.parallel())
+  }
+
+  fun globalSearch(prisonerSearchRequest: PrisonerSearchRequest): Mono<PagePrisoner> {
+    return prisonerSearchWebClient
+      .post()
+      .uri("/global-search")
+      .accept(MediaType.APPLICATION_JSON)
+      .bodyValue(prisonerSearchRequest.toGlobalSearchRequestBody())
       .retrieve()
-      .bodyToMono()
+      .bodyToMono<PagePrisoner>()
+      .onErrorResume { webClientErrorHandler(it) }
   }
+
+  private fun <API_RESPONSE_BODY_TYPE> webClientErrorHandler(exception: Throwable): Mono<API_RESPONSE_BODY_TYPE> =
```

Comment:
I've moved the WebClient error handling here and made it generic so it can be used by /match-prisoners and /global-search (and indeed any endpoint we call on prisoner-offender-search)